qcodes/instrument_drivers/Keysight/KtMAwg.py

```diff
@@ -0,0 +1,319 @@
+
+class KtMAwg(Instrument):
+    _default_buf_size = 256
+    _dll_loc = r"C:\Program Files\IVI Foundation\IVI\Bin\KtMAwg_64.dll"
+
+    def __init__(self, name: str, address: str, options: bytes = b"", **kwargs) -> None:
+        super().__init__(name, **kwargs)
+
+        if not isinstance(address, bytes):
+            address = bytes(address, "ascii")
+
+        self._address = address
+        self._session = ctypes.c_int(0)
+        self._dll = ctypes.windll.LoadLibrary(self._dll_loc)
+        self._channel = ctypes.create_string_buffer("Channel1".encode('ascii'))
+
+        for ch_num in [1, 2, 3]:
+            ch_name = "ch{}".format(ch_num)
+            channel = KtMAWGChannel(self, ch_name, ch_num, )
+            self.add_submodule(ch_name, channel)
+
+        self.get_driver_desc = partial(
+            self.get_vi_string, KTMAWG_ATTR_SPECIFIC_DRIVER_DESCRIPTION)
+        self.get_driver_prefix = partial(
+            self.get_vi_string, KTMAWG_ATTR_SPECIFIC_DRIVER_PREFIX)
+        self.get_driver_revision = partial(
+            self.get_vi_string, KTMAWG_ATTR_SPECIFIC_DRIVER_REVISION)
+        self.get_firmware_revision = partial(
+            self.get_vi_string, KTMAWG_ATTR_INSTRUMENT_FIRMWARE_REVISION)
+        self.get_model = partial(
+            self.get_vi_string, KTMAWG_ATTR_INSTRUMENT_MODEL)
+        self.get_serial_number = partial(
+            self.get_vi_string, KTMAWG_ATTR_MODULE_SERIAL_NUMBER)
+        self.get_manufactorer = partial(
+            self.get_vi_string, KTMAWG_ATTR_INSTRUMENT_MANUFACTURER)
+
+        self._connect(options)
+
+        self.connect_message()
+
+    def _connect(self, options):
+        if not isinstance(options, bytes):
+            options = bytes(options, "ascii")
+        status = self._dll.KtMAwg_InitWithOptions(self._address, 1, 1,
+                                                  options, ctypes.byref(self._session))
+        if status:
+            print("connection to device failed! error: ", status)
+            raise SystemError
+
+    def get_idn(self) -> Dict[str, Optional[str]]:
+        """generates the ``*IDN`` dictionary for qcodes"""
+
+        id_dict: Dict[str, Optional[str]] = {'firmware': self.get_firmware_revision(),
+                                             'model': self.get_model(),
+                                             'serial': self.get_serial_number(),
+                                             'vendor': self.get_manufactorer()}
+        return id_dict
+
+    def _catch_error(self, status):
+        if (status == 0):
+            # No error
+            return
+
+        err = ctypes.c_int32(0)
+        err_msg = ctypes.create_string_buffer(256)
+
+        self._dll.KtMAwg_GetError(
+            self._session, ctypes.byref(err), 255, err_msg)
+        print(f"Got dll error num {err.value} msg {err_msg.value}")
+        raise ValueError
+
+    # Query the driver for errors
+
+    def _get_errors(self):
```

Comment:
This is implemented as a private method but I do not find a use of it anywhere in the driver. If you call this on the driver from an external script, please do not implement is as private (rename it as `get_errors`)